src/Traits/HasRoles.php

```diff
@@ -172,22 +172,24 @@ public function syncRoles(...$roles)
     /**
      * Determine if the model has (one of) the given role(s).
      *
-     * @param string|int|array|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection $roles
-     *
+     * @param  string|int|array|\Spatie\Permission\Contracts\Role|\Illuminate\Support\Collection  $roles
+     * @param  string|null  $guard
      * @return bool
      */
-    public function hasRole($roles): bool
+    public function hasRole($roles, string $guard = null): bool
     {
         if (is_string($roles) && false !== strpos($roles, '|')) {
             $roles = $this->convertPipeToArray($roles);
         }
 
         if (is_string($roles)) {
-            return $this->roles->contains('name', $roles);
+            return $guard ? $this->roles->where('guard_name', $guard)->contains('name',
+                $roles) : $this->roles->contains('name', $roles);
```

Comment:
Can you rewrite these ternaries as multi-line for better readability?
```suggestion
            return $guard 
                ? $this->roles->where('guard_name', $guard)->contains('name', $roles) 
                : $this->roles->contains('name', $roles);
```

Reply:
done